### ISSUE TITLE
tsc Error: Debug Failure.

### ISSUE THREAD

**Issue by samreid** [phetsims/chipper#1388]
Before testing #1387, I ran into this problem

```
cd chipper/tsconfig/all
/Users/samreid/apache-document-root/main/chipper/node_modules/typescript/lib/tsc.js:113984
      throw e;
      ^

Error: Debug Failure.
    at visitEachChildOfIndexSignatureDeclaration (/Users/samreid/apache-document-root/main/chipper/node_modules/typescript/lib/tsc.js:83226:13)
    at visitEachChild (/Users/samreid/apache-document-root/main/chipper/node_modules/typescript/lib/tsc.js:83081:33)
    at visitExistingNodeTreeSymbols (/Users/samreid/apache-document-root/main/chipper/node_modules/typescript/lib/tsc.js:48840:16)
    at visitArrayWorker (/Users/samreid/apache-document-root/main/chipper/node_modules/typescript/lib/tsc.js:82893:49)
    at visitNodes2 (/Users/samreid/apache-document-root/main/chipper/node_modules/typescript/lib/tsc.js:82864:19)
    at visitEachChildOfTypeLiteralNode (/Users/samreid/apache-document-root/main/chipper/node_modules/typescript/lib/tsc.js:83272:7)
    at visitEachChild (/Users/samreid/apache-document-root/main/chipper/node_modules/typescript/lib/tsc.js:83081:33)
    at visitExistingNodeTreeSymbols (/Users/samreid/apache-document-root/main/chipper/node_modules/typescript/lib/tsc.js:48840:16)
    at visitNode (/Users/samreid/apache-document-root/main/chipper/node_modules/typescript/lib/tsc.js:82831:19)
    at visitEachChildOfArrayTypeNode (/Users/samreid/apache-document-root/main/chipper/node_modules/typescript/lib/tsc.js:83278:26)

Node.js v18.12.1
```

I'm still using tsc `Version 5.0.4`, and I do have some working copy changes from https://github.com/phetsims/phet-io/issues/1810.  I'll try reverting the working copy changes and see what happens.  Also, precommit hooks are succeeding, perhaps due to a different scope or caching.

**Comment by samreid** [comment]
Same problem after reverting my working copy and pulling latest.

**Comment by samreid** [comment]
The problem does not appear in tsc `Version 5.1.0-dev.20230420` which I got when I asked for `"typescript": "~5.1.0-beta",`.

**Comment by samreid** [comment]
On slack:

@jonathanolson said:

> I was getting that yesterday yes. Haven’t checked today

I replied

> t seems fixed in ~5.1.0-beta , should I upgrade? Or should we go back to 4.9 for a bit?

@jonathanolson said:

> Fine with either. Slight preference for staying on 5 but curious if we’d run into more issues. Thoughts?

I replied:

> If someone is taking RC shas soon, perhaps 4.9? If RC shas are further out, perhaps 5.1 beta?  Or maybe tsc will fix it in the 5.0 series soon.
